File: back/src/main/java/com/back/domain/venue/entity/Venue.java
package com.back.domain.venue.entity;

import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Venue extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long venueId;

    @Column(nullable = false)
    private String venueName;

    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private Long totalSeats;

    private Venue(String venueName, String location, Long totalSeats) {
        this.venueName = venueName;
        this.location = location;
        this.totalSeats = totalSeats;
    }

    public static Venue create(String venueName, String location, Long totalSeats) {
        return new Venue(venueName, location, totalSeats);
    }
}

File: back/src/main/kotlin/com/back/domain/venue/entity/Venue.kt
package com.back.domain.venue.entity

import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Venue(
    @Column(nullable = false)
    var venueName: String,

    @Column(nullable = false)
    var location: String,

    @Column(nullable = false)
    var totalSeats: Long
) : BaseEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val venueId: Long? = null

    companion object {
        fun create(venueName: String, location: String, totalSeats: Long): Venue {
            return Venue(venueName, location, totalSeats)
        }
    }
}
